cranelift-reader/src/parser.rs

```diff
@@ -3255,6 +3261,20 @@ mod tests {
         );
     }
 
+    #[test]
+    fn u8_as_hex() {
+        macro_rules! parse_as_uimm8 {
+            ($text:expr) => {{
+                Parser::new($text).match_uimm8("unable to parse u8")
+            }};
+        }
```

Comment:
I understand that this macro is keeping with the style of the file, but I wonder why it's using a macro at all instead of just a normal helper function. It seems that if a normal function would suffice, that we should just use that.

Honestly this code is simple enough that even just inlining `Parser::new()` would be perfectly fine.

Reply:
Yeah, good point; I went with a nested function.